src/algebra/group_action_hom.lean

```diff
@@ -0,0 +1,259 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+import algebra.group_ring_action
+
+/-!
+# Equivariant homomorphisms
+
+## Main definitions
+
+* `mul_action_hom M X Y`, the type of equivariant functions from `X` to `Y`.
+* `distrib_mul_action_hom M A B`, the type of equivariant additive monoid homomorphisms
+  from `A` to `B`.
+* `mul_semiring_action_hom M R S`, the type of equivariant ring homomorphisms
+  from `R` to `S`.
+
+## Notations
+
+* `X →[M] Y` is `mul_action_hom M X Y`.
+* `A →+[M] B` is `distrib_mul_action_hom M X Y`.
+* `R →+*[M] S` is `mul_semiring_action_hom M X Y`.
+
+-/
+
+universes u v w u₁
```

Comment:
Do we really need all those universe variables? What happens if you put `Type*` everywhere?

Reply:
I am used to using universe variables in lieu of `Type*` because of https://github.com/leanprover-community/lean/issues/214 but it seems like this issue is resolved now.